modules/client-common/src/main/java/org/apache/ignite/internal/client/proto/ClientBinaryTupleUtils.java

```diff
@@ -435,11 +439,20 @@ public static void appendValue(BinaryTupleBuilder builder, ColumnType type, Stri
                     throw new IllegalArgumentException("Unsupported type: " + type);
             }
         } catch (ClassCastException e) {
+            NativeType nativeType = NativeTypes.fromObject(v);
+            // A null is handled separately, so nativeType should not be null.
+            assert nativeType != null;
+
+            NativeTypeSpec actualType = nativeType.spec();
+            NativeTypeSpec expectedType = NativeTypeSpec.fromColumnType(type);
+
             // Exception message is similar to embedded mode - see o.a.i.i.schema.Column#validate
-            throw new IgniteException(PROTOCOL_ERR, "Column's type mismatch ["
-                    + "column=" + name
-                    + ", expectedType=" + type
-                    + ", actualType=" + v.getClass() + ']', e);
+            String error = format(
+                    "Value type does not match expected {} but got {} in column '{}'",
```

Comment:
```suggestion
                    "Value type does not match. Expected {} but got {} in column '{}'",
```

Reply:
It looks more grammatically correct. If it will be fixed, needs to fix and below messages